### PR TITLE
Fix last modified date issue

### PR DESCRIPTION
**Pull request by alflennik**
The example pages include the date that the example last changed, which is sourced from the git history. However in CI builds of the site, the last modified date ends up being the date the build was last done.

The cause seems to be that GitHub Actions uses a shallow clone instead of a full clone with the complete history.

We won't know for sure if this is solution until we merge this PR to main and the offending action runs.

See https://github.com/w3c/wai-aria-practices/issues/255.

### CODE REVIEW

**Comment by mcking65** [comment]
@alflennik  This is a partial fix, and it is a good step forward.

However, waht we'd ultimately like is quite a bit more complicated. The last changed date should be the date of last change for any of the HTML, CSS, or JS that are specific to the example. We only want to consider the files specific to the example, not any of the shared CSS or JS.

For example, for the select-only combox example, we would want to show the date of most recent change for any of the following files in the `content/patterns/combobox/examples` directory:

* combobox-select-only.html      
* css/select-only.css
* js/select-only.js

**Comment by alflennik** [comment]
@mcking65 the date now reflects all the CSS and JS files! @howard-e this is ready for review again.

**Comment by mcking65** [comment]
Fantastic, thank you @alflennik.

File: scripts/pre-build/library/getExampleLastModifiedDate.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    } catch (error) {
+      console.error(
+        `Failed to extract a last-modified date for the file "${sourcePath}"`
+      );
+      throw error;
+    }
+    return date;
+  };
+
+  const dependencyFilePaths = html
```
Comment (howard-e) [inline scripts/pre-build/library/getExampleLastModifiedDate.js:29]: Clever!

**Review by howard-e — APPROVED** [review]
@alflennik looks good to me! I tested against an updated version of `w3c/aria-practices` locally, and I did also see the dates update where expected.

**Pull request merged**